docs/reference/builder.md

```diff
@@ -180,21 +180,28 @@ that set `abc` to `bye`.
 
 ### .dockerignore file
 
-If a file named `.dockerignore` exists in the root of `PATH`, then Docker
-interprets it as a newline-separated list of exclusion patterns. Docker excludes
-files or directories relative to `PATH` that match these exclusion patterns. If
-there are any `.dockerignore` files in `PATH` subdirectories, Docker treats
-them as normal files. 
-
-Filepaths in `.dockerignore` are absolute with the current directory as the
-root. Wildcards are allowed but the search is not recursive. Globbing (file name
-expansion) is done using Go's
-[filepath.Match](http://golang.org/pkg/path/filepath#Match) rules.
-
-You can specify exceptions to exclusion rules. To do this, simply prefix a
-pattern with an `!` (exclamation mark) in the same way you would in a
-`.gitignore` file. Currently there is no support for regular expressions.
-Formats like `[^temp*]` are ignored. 
+If a file named `.dockerignore` exists at the root of the source
+repository, Docker interprets it as a newline-separated list of exclusion
```

Comment:
I think you are probably right that `PATH` is going to confuse people less.  But strictly speaking there is a second concept.  It is the concept of the thing that `PATH`, `URL`, and `-` all give ways of specifying.  It is what the dockerignore mechanism operates on to produce the build context.  It *is* the build context when there is no `.dockerignore`.

Since introducing a distinct concept is too confusing, the best options I can see are either to go back to using `PATH` or to say that `.dockerignore` contains instructions that the client uses to *modify* the build context before delivering it to the daemon.